class-based/src/say-function.ts

```diff
@@ -0,0 +1,22 @@
+import { Construct } from "constructs";
+import { Function } from "functionless";
+
+interface InputEvent {
+  message: string;
+}
+
+export class SayFunction extends Construct {
+  fn: Function<InputEvent, void>;
```

Comment:
I found this particularly painful with all 3 cdk-leaning approaches. Because the function needs to be passed in, instead of being referred to directly, the user has to create appropriate types and cannot benefit from direct type inference.